tests/e2e/specs/frontend/cart.test.js

```diff
@@ -114,10 +131,12 @@ describe( `${ block.name } Block (frontend)`, () => {
 
 		// This is to ensure we've clicked the right cart button.
 		expect( selectedValue ).toBeGreaterThan( 1 );
-
 		await scrollTo( '.wc-block-cart__submit-button' );
-		await page.click( '.wc-block-cart__submit-button' );
-		await page.waitForSelector( '.wc-block-checkout' );
+		await Promise.all( [
+			page.waitForNavigation(),
+			page.click( '.wc-block-cart__submit-button' ),
+		] );
+		await page.evaluate( () => window.find( 'Checkout Block' ) );
```

Comment:
What is this line for? I understand a value is returned by `page.evaluate` but it isn't being assigned to anything.

Maybe it would be better to use [`page.waitForFunction`](https://pptr.dev/#?product=Puppeteer&version=v8.0.0&show=api-pagewaitforfunctionpagefunction-options-args) here? It would keep retrying until the function returns true if the goal is to wait for the `Checkout Block` text to be on the page?

Reply:
Nice catch Thomas, the assertion was missing here. I change a bit the code to actually test that the h1 on the page contains the text `Checkout`. For some reason on the CI the title is `Checkout` and on the local env the title is `Checkout block`. We will need to investigate why the block navigates to different pages on different envs, but for this PR the purpose is to pass the test.